### PR TITLE
Speech Migration Guide

### PR DESCRIPTION
**Pull request by blowmage**
Because the next release is a breaking change, add a section explaining this to the users and directing them to the migration guide.

### CODE REVIEW

File: google-cloud-speech/google-cloud-speech.gemspec [hunk 1]
```diff
@@ -12,6 +12,12 @@ Gem::Specification.new do |gem|
   gem.homepage      = "https://github.com/googleapis/googleapis"
   gem.license       = "Apache-2.0"
 
+  gem.post_install_message =
+    "This release, 0.30.0, introduces breaking changes relative to the "\
+    "previous release, 0.29.0. For more details and instructions to migrate "\
+    "your code, please visit the migration guide: "\
+    "https://cloud.google.com/speech/docs/ruby-client-migration."
```
Comment (frankyn) [inline google-cloud-speech/google-cloud-speech.gemspec]: Not yet published but will be at: https://cloud.google.com/speech-to-text/docs/ruby-client-migration

Comment (blowmage) [reply google-cloud-speech/google-cloud-speech.gemspec]: K, updated the URL. Thanks!

**Comment by blowmage** [comment]
Once this PR gets approved and merged the master docs will rebuild and we can verify the right code examples are shows for `streaming_recognize`.

File: google-cloud-speech/README.md [hunk 2]
```diff
@@ -19,6 +19,13 @@ steps:
 $ gem install google-cloud-speech
 ```
 
+### Migration Guide
+
+This release, 0.30.0, introduces breaking changes relative to the previous
```
Comment (quartzmo) [inline google-cloud-speech/README.md]: I would recommend future-proofing this sentence with: "The 0.30.0 release introduced ..."

File: google-cloud-speech/google-cloud-speech.gemspec [hunk 3]
```diff
@@ -12,6 +12,12 @@ Gem::Specification.new do |gem|
   gem.homepage      = "https://github.com/googleapis/googleapis"
   gem.license       = "Apache-2.0"
 
+  gem.post_install_message =
+    "This release, 0.30.0, introduces breaking changes relative to the "\
```
Comment (quartzmo) [inline google-cloud-speech/google-cloud-speech.gemspec]: I would recommend future-proofing this sentence with: "The 0.30.0 release introduced ..."

File: google-cloud-speech/lib/google/cloud/speech.rb [hunk 4]
```diff
@@ -35,6 +35,13 @@ module Cloud
     # 3. [Enable the Google Cloud Speech API.](https://console.cloud.google.com/apis/api/speech)
     # 4. [Setup Authentication.](https://googlecloudplatform.github.io/google-cloud-ruby/#/docs/google-cloud/master/guides/authentication)
     #
+    # ### Migration Guide
+    #
+    # This release, 0.30.0, introduces breaking changes relative to the previous
```
Comment (quartzmo) [inline google-cloud-speech/lib/google/cloud/speech.rb]: I would recommend future-proofing this sentence with: "The 0.30.0 release introduced ..."

File: google-cloud-speech/README.md (hunk 2 above)
Comment (blowmage) [reply google-cloud-speech/README.md]: Isn't the intention to remove the warning in the 0.31.0 release? That's what we did for Natural Language.

Comment (quartzmo) [reply google-cloud-speech/README.md]: Maybe so, but future-proofed phrasing is a good way to go just in case it doesn't happen.

Comment (frankyn) [reply google-cloud-speech/README.md]: Given the time-frame of changes for NL, what was the time delta between 0.30.0 and 0.31.0?

**Pull request merged**